kinto/plugins/quotas/listener.py

```diff
@@ -0,0 +1,212 @@
+import copy
+
+from pyramid.httpexceptions import HTTPInsufficientStorage
+from kinto.core.errors import http_error, ERRORS
+from kinto.core.storage.exceptions import RecordNotFoundError
+from kinto.core.utils import instance_uri
+
+from .utils import record_size
+
+
+def get_bucket_settings(settings, bucket_id, name):
+    return settings.get(
+        # Bucket specific
+        'quotas.bucket_{}_{}'.format(bucket_id, name),
+        # Global to all buckets
+        settings.get('quotas.bucket_{}'.format(name), None))
+
+
+def get_collection_settings(settings, bucket_id, collection_id, name):
+    return settings.get(
+        # Specific for a given bucket collection
+        'quotas.collection_{}_{}_{}'.format(bucket_id, collection_id, name),
+        # Specific to given bucket collections
+        settings.get('quotas.collection_{}_{}'.format(bucket_id, name),
+                     # Global to all buckets collections
+                     settings.get('quotas.collection_{}'.format(name), None)))
+
+
+def on_resource_changed(event):
+    """
+    Everytime an object is created/changed/deleted, we update the
+    bucket counters. The entries are served as read-only in the
+    :mod:`kinto.plugins.quotas.views` module.
+
+    If a new object override the quotas, we reject the request.
+    """
+    payload = copy.deepcopy(event.payload)
```

Comment:
You are right, it is a left over from the history plugin I took the idea from. No need for that here.